packages/hardhat-core/src/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -226,6 +234,39 @@ export class HardhatModule {
     return true;
   }
 
+  // hardhat_setStorageSlot
+
+  private _setStorageSlotParams(params: any[]): [Buffer, BN, Buffer] {
+    const validatedParams = validateParams(
+      params,
+      rpcAddress,
+      rpcQuantity,
+      rpcData
+    );
+
+    const MAX_WORD_VALUE = new BN(2).pow(new BN(256));
+    if (validatedParams[1].gt(MAX_WORD_VALUE)) {
+      throw new InvalidInputError("Storage key must not be greater than 2^256");
+    }
+
+    if (validatedParams[2].length !== 32) {
+      throw new InvalidInputError(
+        "Storage value must be exactly 32 bytes long"
+      );
+    }
```

Comment:
Hmm, maybe? But it seems like they are intentionally not checking that (plus it would be a breaking change), so I don't know if they'd be willing to do it.